deathcause.lua

```diff
@@ -0,0 +1,212 @@
+                table.insert(items, item)
+            end
+        end
+    end
+    return items
+end
+
+-- Finds a unit with the given id or nil if no unit is found
+function unit_find(unit_id)
+    return utils.binsearch(df.global.world.units.all, unit_id, 'id')
+end
+
+function getRace(race_id)
+    return df.global.world.raws.creatures.all[race_id]
+end
+
+function getRaceNameSingular(race_id)
+    return getRace(race_id).name[0]
+end
+
+function getDeathStringFromCause(cause)
+    if cause == -1 then
+        return "none"
+    else
+        return DEATH_TYPES[cause]
+    end
+end
+
+function display_death_unit(unit)
+    local str = string.format("The %s", getRaceNameSingular(unit.race))
+    if unit.name.has_name then
+        str = str .. string.format(" %s", dfhack.TranslateName(unit.name))
+    end
+
+    if not unit.flags2.killed and not unit.flags3.ghostly then
+        str = str .. " is not dead yet!"
+        print(str)
+        return
+    end
+
+    str = str .. string.format("%s", getDeathStringFromCause(unit.counters.death_cause))
+
+    local incident = df.incident.find(unit.counters.death_id)
+    if incident then
+        str = str .. string.format(" in year %d", incident.event_year)
+    end
+
+    if incident and incident.criminal then
+        local killer = df.unit.find(incident.criminal)
+        if killer then
+            str = str .. string.format(" killed by the %s", getRaceNameSingular(killer.race))
+            if killer.name.has_name then
+                str = str .. string.format(" %s", dfhack.TranslateName(killer.name))
+            end
+        end
+    end
+
+    str = str:gsub(",%s*$", "")
+    print(str)
+end
+
+function display_histfig_death_event(histfig_unit, event)
+    local str = getUnitDeathString(histfig_unit);
+    local str = string.format(
+            "The %s %s %s in year %d,",
+            getRaceNameSingular(histfig_unit.race),
+            dfhack.TranslateName(histfig_unit.name),
+            getDeathStringFromCause(event.death_cause),
+            event.year
+    )
+
+    if event.slayer_hf and event.slayer_hf ~= -1 then
+        local slayer_histfig = utils.binsearch(df.global.world.history.figures, event.slayer_hf, 'id')
+        if slayer_histfig then
+            str = str .. string.format(
+                    " killed by the %s %s",
+                    getRaceNameSingular(slayer_histfig.race),
+                    dfhack.TranslateName(slayer_histfig.name)
+            )
+        end
+    end
+
+    if event.weapon then
+        if event.weapon.item_type == df.item_type.WEAPON then
+            str = str .. string.format(" using a %s", df.global.world.raws.itemdefs[event.weapon.item_subtype].name)
+        elseif event.weapon.shooter_item_type == df.item_type.WEAPON then
+            str = str .. string.format(", shot by a %s", df.global.world.raws.itemdefs[event.weapon.shooter_item_subtype].name)
+        end
+    end
+
+    str = str:gsub(",%s*$", "")
```

Comment:
again, I'm not seeing where this can happen

Reply:
This _could_ happen if the event does not have a slayer and the event does not have a weapon. I don't know enough about DF to say whether that can actually happen in the real world or not (I just started playing with the steam release).

Reply:
I am however going to change it to have the other strings prepend the command instead of that first string appending it